bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/BookieProtoEncoding.java

```diff
@@ -309,6 +367,22 @@ public Object decode(ByteBuf buffer)
 
                 return new BookieProtocol.ReadResponse(
                         version, rc, ledgerId, entryId, buffer.retainedSlice());
+            case BookieProtocol.BATCH_READ_ENTRY:
+                rc = buffer.readInt();
+                ledgerId = buffer.readLong();
+                entryId = buffer.readLong();
+                long requestId = buffer.readLong();
+                ByteBufList data = null;
+                while (buffer.readableBytes() > 0) {
+                    int entrySize = buffer.readInt();
+                    if (data == null) {
+                        data = ByteBufList.get(buffer.readBytes(entrySize));
```

Comment:
Can we use `buffer.retainedSlice()` to avoid data copy?

Reply:
make sense.